packages/testing/src/cli.ts

```diff
@@ -174,6 +176,14 @@ export async function main(argv = process.argv) {
             console.log();
         }
     }
+
+    // Do not hang indefinitely if tests do not clean up after themselves properly.  Instead print error and exit with
+    // non-zero status code
+    const timeout = setTimeout(() => {
+        console.error(`Error: Tests passed but process did not exit cleanly after ${SHUTDOWN_TIMEOUT_MS / 1000}s.`);
+        process.exit(101);
```

Comment:
Idea: could we trigger logging node diagnostics in such a case? to directly "see" whats up?

Reply:
That would be good, I wanted to dump diagnostic information but I don't know of anything that would be useful, do you?

What we really need is wtfnode but that would require us to run all tests with wtf enabled. Which would maybe work I just worry about perf/stability because it's so invasive.

Reply:
I would log the same as SIGUSR2 would do for the start

Reply:
Will address when it comes up next